pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,147 @@
 """Tests that sync container plugin repositories."""
-from pulp_smash import config
-from pulp_smash.pulp3.bindings import (
-    delete_orphans,
-    monitor_task,
-    PulpTaskError,
-    PulpTestCase,
-)
-from pulp_smash.pulp3.utils import gen_repo
-from pulp_container.tests.functional.utils import (
-    gen_container_client,
-    gen_container_remote,
-)
+import pytest
+from pulpcore.tests.functional import PulpTaskError
+
+from pulp_container.tests.functional.utils import gen_container_remote
 from pulp_container.tests.functional.constants import PULP_FIXTURE_1
 
-from pulpcore.client.pulp_container import (
-    ContainerContainerRepository,
-    ContainerContainerRemote,
-    ContainerRepositorySyncURL,
-    ContentTagsApi,
-    RepositoriesContainerApi,
-    RepositoriesContainerVersionsApi,
-    RemotesContainerApi,
-)
+from pulpcore.client.pulp_container import ContainerContainerRemote
 
 
-class BasicSyncTestCase(PulpTestCase):
-    """Sync repositories with the container plugin."""
+@pytest.fixture
+def synced_repository_with_filtered_tags(
+    container_repo, container_remote_api, gen_object_with_cleanup, sync
+):
+    def _synced_repository_with_filtered_tags(include_tags=None, exclude_tags=None):
+        """Sync a new repository with the included tags passed as an argument."""
+        remote_data = gen_container_remote(
+            upstream_name=PULP_FIXTURE_1,
+            include_tags=include_tags,
+            exclude_tags=exclude_tags,
+        )
+        remote = gen_object_with_cleanup(container_remote_api, remote_data)
 
-    maxDiff = None
+        sync(remote, container_repo)
 
-    @classmethod
-    def setUpClass(cls):
-        """Create class-wide variables."""
-        cls.cfg = config.get_config()
-        cls.client_api = gen_container_client()
+        return container_repo
 
-    def test_sync(self):
-        """Sync repositories with the container plugin.
+    return _synced_repository_with_filtered_tags
 
-        In order to sync a repository a remote has to be associated within
-        this repository. When a repository is created this version field is set
-        as None. After a sync the repository version is updated.
 
-        Do the following:
+@pytest.mark.parallel
+def test_basic_sync(container_repo, container_remote, container_repository_api, sync):
+    assert container_repo.latest_version_href == f"{container_repo.pulp_href}versions/0/"
```

Comment:
To avoid what this assertion seems to fail on i would use a factory again.

Reply:
same here. :(

Reply:
I am pleading for a factory that creates a new randomly named empty repository. No need to assert, that it is empty...

Reply:
We can remove this redundant assertion.